### DISCUSSION TITLE
Is gradual Migration from Stencil to Catalyst possible?

### DISCUSSION THREAD

**Discussion by deanklijbroek** [bigcommerce/catalyst#1138] [Q&A] [answered]
We are currently running our website on the Stencil framework and are considering a gradual migration to Catalyst. We do not want to transition all at once, but rather replace our existing Stencil pages with Catalyst pages incrementally. so this means we want to run a 'hybride' model where we both use Stencil and Catalyst at the same time.

So, for  example: 

When reaching a category listing page: Stencil page
When reaching a product detail page: Catalyst page

Is it possible to use both Stencil and Catalyst simultaneously within the same BigCommerce project? If so, are there specific best practices or technical requirements we should be aware of during this partial migration? Are there any limitations or issues we might encounter when using both frameworks concurrently (e.g. cart line items). 

Thanks in advance!

**Comment by jorgemoya** [comment, marked as answer]
Good question!  Unfortunately, it's not possible at the moment, but we are definitely thinking about migration solutions and will provide updates if we have any new details.

**Reply by z0n** [reply]
Is there any update on this? We're about to start with a new headless store and would like to migrate page by page.

We were thinking about using Next.js rewrites but there might be some challenges with session syncing (cart etc.).

**Reply by Healthydad007** [reply]
It's been a year, any update to this? Perhaps a guide for store owners on best practices to migrate?

**Comment by Healthydad007** [comment]
Any update to this yet? I dont understand how BigCommerce has not provided a solution for stores running on stencil to migrate....
